src/shogun/base/SGObject.h

```diff
@@ -533,11 +533,13 @@ class CSGObject
 	 */
 	virtual std::string to_string() const;
 
-	/** Returns set of all parameter names of the object.
+	/** Returns map of parameter names and AnyParameter pairs
+	 * of the object.
 	 *
 	 */
-	std::vector<std::string> parameter_names() const;
-
+#ifndef SWIG // SWIG should skip this part
+	std::map<std::string, std::shared_ptr<const AnyParameter>> get_params() const;
```

Comment:
doxygen string would be good, like what is this method doing.

Reply:
Oh it's already there but I put the `#ifndef SWIG` bellow it. I guess it should go above the oxygen string?

Reply:
ah sorry I didnt see